internal-dns/src/resolver.rs

```diff
@@ -802,7 +790,7 @@ mod test {
         let dns_server = DnsServer::create(&logctx.log).await;
         let resolver = Resolver::new_from_addrs(
             logctx.log.clone(),
-            vec![*dns_server.dns_server.local_address()],
+            &[*dns_server.dns_server.local_address()],
```

Comment:
Similar nit here - I think `local_address()` should probably return `SocketAddr` instead of `&SocketAddr`